BMS_Receiver/bms_Rx_adapfunc.c

```diff
@@ -0,0 +1,171 @@
+
+float getParamValuefromString(char *scanLine, enum BATTERYPARAM batteryParam)
+{
+  char splitStr[NUMOFPARAM*2][12]={'\0'};
+  char * pch;    
+  int k=0;
+  float temp;
+  char buffer[MAXLENGTH_INPUTSTRING];
+  
+  strcpy(buffer,scanLine);
+  pch = strtok (buffer," :\t");
+  
+  while (pch != NULL)
+  {
+    
+    strcpy((splitStr[k]),pch);
+	//printf("\n%s",splitStr[k]);
+    k++;
+    pch = strtok (NULL, " :\t");
+  } 
+  
+  for(int j=0;j < (NUMOFPARAM*2);j=j+2)
+  {
+        
+      if(strcmp((splitStr[j]),BatteryParam[batteryParam].ParamName) == 0)
+      {
+         temp = atof(splitStr[j+1]);
+	
+         break;
+      }
+      else
+      {
+          temp = VALUE_NOTFOUND;
+      }
+  }
+
+  pch = NULL;
+
+  return temp; 
+}   
+
+
+/****************************************************************************************
+*Func desc : The function which updates the calculated value based on passed structure, it can be adapted based on needs, 
+			 Currently it does printing of Temperature data in the following format 
+			 "Temperature -> SMA : 14.08 degrees, Minimum : 70.41 degrees , Maximum : 70.41 degrees"
+			 If additional parameters are to be added as output , the BatteryParamEvaluated needs to be adapted followed by printf realisation
+*Param     : BatteryParamEvaluated  - structure which holds the calculated value
+			 .SMA       - Calculated SMA value
+			 .minRxd    - The valid minimum value received over console
+			 .maxRxd    - The valid maximum value received over console
+*Return    : Updation status- bool type
+*****************************************************************************************/
+
+bool UpdateTemperatureCalcData (struct BatteryParamOutput_s BatteryParamEvaluated)
+{
+	 /*This function can be adapted in production based on the usage, say to append data to file or Printing as needed*/
+    #if(TEST_MODE)
+    int i=TestCount[TEMPERATURE];
+    TestSMAOuputValue[i][TEMPERATURE] = BatteryParamEvaluated.SMA;
+    TestMinOuputValue[i][TEMPERATURE] = BatteryParamEvaluated.minRxd;
+    TestMaxOuputValue[i][TEMPERATURE] = BatteryParamEvaluated.maxRxd;
+	TestCount[TEMPERATURE]++;
+	if(TestCount[TEMPERATURE] >= TestSize)
+	{
+		sig_caught = 1;
+	}
+	else
+	{
+		sig_caught = 0;
+	}
+    #endif
+	if (BatteryParamEvaluated.SMA == INVALID_VALUE)
+	{
+	   printf("No Valid Temperature data received\n");
+	}
+	else
+	{
+		printf("Temperature -> SMA : %.2f degrees, Minimum : %.2f degrees , Maximum : %0.2f degrees \n", BatteryParamEvaluated.SMA, BatteryParamEvaluated.minRxd, BatteryParamEvaluated.maxRxd);
+	}
+		
+    return 1;
+}
+
+/****************************************************************************************
+*Func desc : The function which updates the calculated value based on passed structure, it can be adapted based on needs, 
+			 Currently it does printing of charge rate data in the following format 
+			 "ChargeRate -> SMA : 0.84 A , Minimum : 4.20 A, Maximum : 4.20 A"
```

Comment:
Excellent comments.
See if some of the details can be moved to tests. Because if the output or language of output changes, this will need to be updated and maintainers are likely to forget. If it's asserted in a test, it would fail and the test would be updated.